src/main/java/cat/udl/eps/softarch/myroutes/domain/RouteVersion.java

```diff
@@ -1,24 +1,37 @@
 package cat.udl.eps.softarch.myroutes.domain;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.Id;
+import jakarta.persistence.ManyToOne;
 import jakarta.validation.constraints.NotBlank;
 import jakarta.validation.constraints.NotNull;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 
 import java.time.ZonedDateTime;
 
 @Entity
+@EqualsAndHashCode(callSuper = true)
 @Data
 public class RouteVersion extends UriEntity<Long>{
 
     @Id
-    @GeneratedValue
+    @GeneratedValue()
     private Long id;
 
-    @NotBlank
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
     @NotNull
     private ZonedDateTime creationDate;
 
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    @ManyToOne
+    @NotNull
+    private User createdBy;
+
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
```

Comment:
If `versionOf` is set to READ_ONLY, it cannot be set through the API. Therefore, right now it's quite likely that this field remains `null` when a route version is created or updated. An alternative would be to make it writable but control in the handler that it cannot be changed after a route version is created.